src/sortStudents.ts

```diff
@@ -1,16 +1,106 @@
+// import { type } from "os";
 
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name,
+  Surname,
+  Age,
+  Married,
+  AverageGrade,
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: string,
+): Student[] {
+  const copyStudents: Student[] = [...students];
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+  type Sum = (sum: number, mark: number) => number;
+
+  const sumMarks: Sum = (sum, mark) => sum + mark;
+
+  switch (sortBy) {
+    case SortType.Name:
+      if (order === 'asc') {
+        copyStudents.sort(
+          ({ name: nameA }, { name: nameB }) => nameA.localeCompare(nameB),
+        );
+      } else {
+        copyStudents.sort(
+          ({ name: nameA }, { name: nameB }) => nameB.localeCompare(nameA),
+        );
+      }
+      break;
+
+    case SortType.Surname:
+      if (order === 'asc') {
+        copyStudents.sort((
+          { surname: surnameA },
+          { surname: surnameB },
+        ) => surnameA.localeCompare(surnameB));
+      } else {
+        copyStudents.sort((
+          { surname: surnameA },
+          { surname: surnameB },
+        ) => surnameB.localeCompare(surnameA));
+      }
+      break;
+
+    case SortType.Age:
+      if (order === 'asc') {
+        copyStudents.sort(({ age: ageA }, { age: ageB }) => ageA - ageB);
+      } else {
+        copyStudents.sort(({ age: ageA }, { age: ageB }) => ageB - ageA);
+      }
+      break;
+
+    case SortType.Married:
+      if (order === 'asc') {
+        copyStudents.sort((
+          { married: marriedA },
+          { married: marriedB },
+        ) => Number(marriedA) - Number(marriedB));
+      } else {
+        copyStudents.sort((
+          { married: marriedA },
+          { married: marriedB },
+        ) => Number(marriedB) - Number(marriedA));
+      }
+      break;
```

Comment:
You can create a variable `sortOrder`, this variable would be `1` in `asc` order or `-1` for `desc`
```suggestion
    case SortType.Married:
        copyStudents.sort((
          { married: marriedA },
          { married: marriedB },
        ) => (Number(marriedA) - Number(marriedB)) * sortOrder);
        
      break;
```

Reply:
it's brilliant advice! thx! I know that sort method works with -1, 0, 1, but I didn't guess that I can use a variable to regulate it 🔥